Add section about OpenID Connect in docs

jkakavas opened the pull request:
Attempted to consolidate common/similar information for SSO related auth providers in order to minimize duplication but I'm really interested in feedback in case this became very complicated / unclear.

alexwizp committed:
TSVB:  selecting the colour picker crashes Kibana (#37176)

Fix: #37115

jkakavas committed:
Add section about OpenID Connect in docs

elasticmachine commented:
Pinging @elastic/kibana-docs

elasticmachine commented:
Pinging @elastic/kibana-security

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/15528/)

vitalics committed:
Tsfy screenshot service and comparepng internal lib (#37177)

vitalics committed:
[TSVB] markdown variables(revert fix) (#34067)

* add markdown tests

rashidkpc committed:
Fix Canvas fullscreen

Closes #34080

For whatever reason, `Style.it` is unwrapping and dropping the root node of the component it wraps. No idea why. This simply wraps the tree up in a sacrificial `<div/>` for `Style.it` to eat.

Commit:
Revert "Fix Canvas fullscreen"

This reverts commit f02c78b8b62a0941ffa980f0f5bc6b3c037b9c53

ogupte committed:
Closes #36992 by fixing path params bug by checking if serviceName even (#37062)

exists in the current pathname

kertal committed:
Add  automatic alignment to gauge visualisation (#34845)

Add flexibility to alignment of multiple gauge vis by supporting the following alignments: automatic, horizontal, vertical. the default is: automatic, it aligns the gauges depending on screen space to horizontal or vertical

Further Adaptions
- sample data
- migration of saved objects
- removal of jquery as depencency
- adaptions of rendering to prevent clipping

dmlemeshko committed:
TSVB: functional test for color picker, preview disable and series clone (#37186)

* [visualize/_tsvb_chart] test for color picker, changing  preview and cloning series

walterra committed:
[ML] Data Frames: Fix to avoid queries for indices with more than 1024 fields. (#37183)

- 7.0 introduced indices.query.bool.max_clause_count which defaults to 1024. This can break certain queries (e.g. simple_query) for indices which have more than 1024 fields (e.g. certain beats indices). The optional data frames query uses simple_query and could therefor break the source index preview as well the pivot preview and pivot job itself given these conditions.
- Originally the default query (* used for simple_query) was always applied for source index previews and pivot previews. A new check isDefaultQuery() will now allow a) the source index preview to use a more efficient match_all query and b) avoid adding the query to the pivot config. This avoids triggering the max_clause_count when no optional query is set.
- If an index has more than 1024 fields, the input form for an optional query will be hidden. A helper text explains the reasoning. This avoids triggering max_clause_count related errors from within the UI. A user can still copy a UI created config to the clipboard and add an optional query in Kibana dev console.
- Additionally, this PR adds a fix to format date fields in the source index preview table using moment-timezone and formatHumanReadableDateTimeSeconds to display dates with the correct timezone.

sulemanof committed:
Move all switches to the SwitchParamEditor (#37215)

kobelb committed:
[docs][skip ci] Infrastructure and Logs read only access (#35566)

* Adding a read only access section to the infrastructure docs

* Adjusting where the infrastructure read only access section is and
adding the logs readonly access section

* Fixing grammar

Commit:
Moved query filter and push filter into filter manager (#36764)

* Moved query filter and push filter from filter bar to filter manager

walterra committed:
[ML] Fix access denied for data frames. (#37178)

With insufficient privileges, a user would be redirected to the ML plugin's access-denied page which mentions the required user roles to access the pages. Since data frames introduces new user roles these messages were not correct.

This PR fixes it by redirecting to a specific access-denied page for data frames. To avoid to much refactoring as a fix, the page is a copy and port to React of the original one. In a follow up for 7.3, we should merge the two pages and it should have options to display required user roles given a certain context like anomaly detection or data frames.

lcawl committed:
[DOCS] Facilitates content re-use in the monitoring configuration (#37123)

walterra committed:
[ML] Fix to not add configs with names which would result in nesting conflicts (#37212)

Two group-by or aggregation names could result in conflicting nested fields which would return an error for pivot previews. For example, two names like responsetime and responsetime.avg are not allowed.

This PR fixes the issue by extending the tests whether a configuration is allowed to be added to the list of group-by and aggregation configurations. If a conflict is detected, a toast notification gets triggered and the configuration won't be added.

andrew-goldstein committed:
[SIEM] Fixes timeline notes overflowing the modal (#37134)

## [SIEM] Fixes timeline notes overflowing the modal (#37134)

Beginning in Chrome 74 (see elastic/eui#1902), the timeline notes modal can overflow in some browsers, per the _Before_ gif below.

This fix uses `EuiModal` and other `EUI` components to address the overflow issue across all the browsers tested in the gifs below:

### Before (Chrome `74.0`)

![01-before-chrome](https://user-images.githubusercontent.com/4459398/58355997-4d7d4d00-7e33-11e9-864b-7e77d0635116.gif)

### After (Chrome `74.0`)

![02-after-chrome-74 0](https://user-images.githubusercontent.com/4459398/58356028-6128b380-7e33-11e9-8c7d-3022e45b1f41.gif)

### After (Firefox `67.0`)

![03-after-firefox-67 0](https://user-images.githubusercontent.com/4459398/58356056-6ede3900-7e33-11e9-92ca-a6dd8e0b804b.gif)

### After (Safari `12.1.1`)

![04-after-safari-12 1 1](https://user-images.githubusercontent.com/4459398/58356123-a947d600-7e33-11e9-80ab-d6b3d3c601c0.gif)

### After (IE`11.0.9600`)

![05-after-ie-11](https://user-images.githubusercontent.com/4459398/58356131-aea52080-7e33-11e9-989e-51aab7c4e9da.gif)

https://github.com/elastic/ingest-dev/issues/442

alvarezmelissa87 committed:
return false for each jobId if it does not exist (#37116)

FrankHassanabad committed:
[SIEM] Low impact linter rules to start with (#37137)

## Summary

Low impact linter rules to help catch bugs for back-porting as well as moving into the future. We will slowly turn the warns into errors and fix the errors as we go along. This helps us avoid a "big bang" switch over approach to avoid back-port issues as we are porting bugs aggressively at the beginning.

Added a custom pattern detection rule for determining if the UI is trying to import anything from the server backend since that would cause web-pack front to pull in backend code. Also forbid the use of NodeJS imports on the front end.  

Double checked that I was not duplicating rules from:
* https://github.com/elastic/kibana/blob/master/packages/eslint-config-kibana/typescript.js
* https://github.com/elastic/kibana/blob/master/packages/eslint-config-kibana/javascript.js

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~
~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

XavierM committed:
fix persistence of timeline when you pinned or add a note to an event (#37248)

FrankHassanabad committed:
[SIEM] Fixed unescaped HTML entity issue (#37257)

## Summary
  * Turned on linter rule to be an error
  * Removed the extra character
  * https://github.com/elastic/ingest-dev/issues/467

Removes the extra `}` character seen after line 84 below in the screen shot:

<img width="368" alt="Screen Shot 2019-05-28 at 9 39 28 AM" src="https://user-images.githubusercontent.com/1151048/58498271-f4672f00-813a-11e9-8b80-2326db363bb0.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

alvarezmelissa87 committed:
[ML] Prevent overflow of large metric values in data visualizer/file data visualizer cards (#36932)

* filedataviz cards - abbrev long numeric values

* add directive for index dataviz values

* Add arg to indicate format. update tests

* truncate with ellipsis

* use eui ellipsis truncation

* increase card content height

* ensure truncation for firefox and safari

Commit:
update yarn.lock file

FrankHassanabad committed:
[SIEM] Fix state issues to avoid potential React update issues (#37261)

## Summary

Fixed prevState issues with React setState
  * This will avoid subtle bugs with the react update renderer
  * Enabled one lint rule from warn to error
  * Turned off the other linter issue as it has an open ticket against it.
  * https://github.com/elastic/ingest-dev/issues/468

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

mshustov committed:
Expose plugin contracts to Legacy platform (#37218)

* expose plugin contracts to Legacy platform

* move mapToObject to src/core/utils

dmlemeshko committed:
[visualize/_shared_item] skip test due to #37130 (#37131)

Bargs committed:
Fix infinite loop in QueryBarInput's componentDidUpdate (#36946)

We were calling updateSuggestions unconditionally in componentDidUpdate
which in turn called setState unconditionally, so componentDidUpdate
was running in an infinite loop.

kobelb commented on docs/security/authentication/index.asciidoc:
```diff
@@ -80,32 +81,93 @@ IMPORTANT: The {kib} user-facing origin should be the same in {kib}, {es}, and t
-xpack.security.authProviders: [saml, basic]
+xpack.security.authProviders: [oidc, basic]
 --------------------------------------------------------------------------------
 --
```
```suggestion
```
I don't know how it worked previously, but I'm unable to build the docs using `docbldkb` from https://github.com/elastic/docs/blob/master/doc_build_aliases.sh#L26 without deleting this line... 🤷‍♂

w33ble committed:
Fix: Fullscreen sizing on old workpads (#37289)

* fix: provide css value for workpads

old workpads were missing this value, this fixes scaling in fullscreen mode

* fix: make workpadCss a required prop

this will cause warnings in the future if the value is missing. also remove the unused css prop.

mw-ding committed:
[Code] Reenable lsp_indexer mocha tests and incooporate language filter test cases (#37266)

chongzhe committed:
[Code] change Typescript to TypeScript in UI (#37270)

jkakavas committed:
Remove validation from OIDC endpoint (#37159)

* Remove validation from OIDC endpoint

This change removes validation of query parameters from the
/api/security/v1/oidc endpoint. It was discovered during manual
testing that Google's OP is sending extra parameters than the ones
identified in https://tools.ietf.org/html/rfc6749#section-4.1.2
which is refernced by
https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse
(for instance auth_user and session_state). The existing validation
rules only allowed the expected query parameters but this
means that Kibana wouldn't be able to complete OpenID Connect
authentication with Google acting as the OP.
As dictated in the standard (RFC6749), "The client MUST ignore
unrecognized response parameters." so we should allow but discard
any extra parameters we do not recognize and not throw an error.

* address feedback

* address feedback

* add validation for the issuer

jkakavas committed:
Apply suggestions from code review

Co-Authored-By: Brandon Kobel <brandon.kobel@gmail.com>

kertal committed:
Remove duplicate fields of rollup based index patterns (#37221)

Rollup fields were were merged twice with other fields, this caused duplicates. Fix of a bug introduced by #28762

angorayc committed:
[SIEM] Replace Eui chart with elastic charts for siem kpis (#36660)

* move charts to separate components

* replace areachart

* apply custom styles

* customize barchart color

* customize color for areachart

* move reusable functions into common

* exchange x & y value in barchart dataset

* replace pure component with react memo and upgrade enzyme adapter

alexwizp committed:
Visual Builder duration in second showing as number (#35813)

* Visual Builder duration in second showing as number

Fix: #35641 , #35805

* fix PR comments

* Visual Builder duration in second showing as number

* Visual Builder duration in second showing as number -fix placeholder

Commit:
Move filter bar and apply filters to data plugin (#36778)

* Setup filter plugin
With filter_bar and ApplyFiltersPopover
Update imports

* Updated i18n keys

* Typo fix

* update JP translation names

azasypkin committed:
Introduce Kerberos authentication provider. (#36112)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/15761/)

maryia-lapata committed:
[Vis Editor] Fix undefined error in Field control (#37179)

* Add default value for labels in sortByLabel function

* TS: make label optional

sulemanof committed:
[Vis: Default editor] EUIficate top_sort param editor (#36629)

* EUIficate top_sort param editor

* Set field if only one available

* Remove unused translations

* Fix functional tests

* Fix browser tests

* Rewrite functional test for combo_box field select

* Remove unused translations

* Update label

vitalics committed:
[TSVB] [AT] TimeSeries refactor (#36987)

* move tsvb timeseries tests into new file

sulemanof committed:
Upgrade enzyme and enzyme-adapter-react-16 to the latest (#37064)

* Upgrade enzyme and enzyme-adapter-react-16 to the latest

* Skip failed expectation

* Update yarn.lock

* Update snapshot

mshustov committed:
add docs how to use config in New Platform (#36782)

* add docs how to use config in NP

* do not encourage storing initializer context in plugin

maryia-lapata committed:
[Vis: Default editor] EUIficate controls of Geohash agg (#37213)

* EUIficate controls of geohash agg

* Fix tests

Commit:
[Code] fix lsp init unit test by using fake timers (#37201)

* [Code] fix LSP init unit test by using fake timers

mshustov committed:
introduce pre-/post-auth request hooks for HttpServer (#36690)

* introduce pre-,post-auth stages

* cleanup integration_tests. now contracts available in tests

* auth per route is configurable

* Unify lifecycle results structure

* expose api to store auth state and status via http service

* update tests

* update docs

* use full name, auth should not mutate request

* move basePath functionality under namespace

* regenerate docs

* Revert "move basePath functionality under namespace"

This reverts commit 9599d328013b0ba734294959b90454eeeb9382f3.

* Revert "regenerate docs"

This reverts commit 1799d3b0881b175ab2d36b312d1172806ab2a113.

* regenerate docs

* updated yarn.lock no idea why

* extract AuthStateStorage to a separate entity

* get rid of nested ifs

* describe what is the difference between hooks

* re-wording

sulemanof committed:
[Vis: Default editor] EUIficate top_aggregate and size param editors (#36567)

* EUIficate top_aggregate_and_size param editor

* Remove template

* Change typescript interfaces

* Fix browser tests

* Add an icon alert

* Changes due to comments

* Move error to a help text

* Move error to a field

* Change validation logic

* Fix discarding changes action

* Remove changed translation

maryia-lapata committed:
[Vis: Default editor] EUIficate Percentile Ranks and Percentiles (#35349)

* Migrate number-list

* Add types

* Refactoring

* Add number row

* Add and remove items

* Add onChange

* Add validate function

* Apply validation

* Fix merge conflict

* Apply order validation

* Fix ts

* Refactoring

* EUIficate percentiles control

* Remove unused translations

* Move helper functions to a separate file

* Fix code review comments

* Make add button 'xs' size

* Remove extra space

* Fix validation

* Remove unused translations

Bamieh committed:
[Docs] i18n docs add japanese (#37233)

sulemanof committed:
[Vis: Default editor] EUIficate order_agg param editor (#36984)

* EUIficate order_agg param editor

* Update browser tests

* Add types

* Update enzyme version in x-pack

* Fix functional tests

* Changes due to comments

* Update_terms_helper.tsx

Co-Authored-By: Maryia Lapata <mary.lopato@gmail.com>

* Fix code review comments

* Update yarn.lock

Bamieh committed:
[i18n] JP translations (#37229)

* updated JP translations

* run i18n check

legrego committed:
Security - remove auth scope provider (#36998)

* remove auth scope provider

* handle missing roles

* guard for unauthenticated calls

* update functional tests to not expect a scope property

* there's always money in the banana stand

* revert interceptor optimizations

* protect against missing roles

* address pr feedback

* remove scope as expected property on kerberos auth response

Commit:
[SIEM] Update Empty Page Messages (#37251)

* update EmptyPage comp and text on all pages

* change proptype from `string` to `IconType`

* update tests and snapshots

* remove translation imports

* restore `align-self` styles for IE11 centering

* proper apostrophe and string in single quotes, per frank

* update host and ip details

* DRYing up host and network empty page message

* fix localization

* fixing localization again

Commit:
[checks-reporter] reenable (#37280)

cqliu1 committed:
[Canvas] Fixes Workpad prop typo (#37351)

Conky5 committed:
Remove directory from dockerfiles target (#37258)

* Remove directory from dockerfiles target

Produce a docker build context tarball without a top-level directory to
match the expected format for downstream automation.

This will produce:

```
$ tar -tzf target/kibana-8.0.0-docker-build-context.tar.gz
Dockerfile
bin/
config/
bin/kibana-docker
config/kibana.yml
```

Instead of:

```
$ tar -tzf target/kibana-8.0.0-docker-build-context.tar.gz
kibana-8.0.0-docker-build-context/Dockerfile
kibana-8.0.0-docker-build-context/bin/
kibana-8.0.0-docker-build-context/config/
kibana-8.0.0-docker-build-context/bin/kibana-docker
kibana-8.0.0-docker-build-context/config/kibana.yml
```

* Use options object for root directory config

Commit:
[Feature/Reporting] Export Saved Search CSV as Dashboard Panel Action (#34571)

XavierM committed:
[SIEM] fix data provider query when the field is a timestamp (#37281)

* fix dataprovider query when the filed is a timestamp

* review II

* review III

timroes committed:
Add addError function to toastNotifications (#32187)

mattkime committed:
bump github-checks-reporter to delay 60s on retries and retry 10x (#37358)

toddself committed:
[feat] create additional http servers (#36804)

* [feat] create additional http servers

allow for additional http servers to be created, tracked and returned

* respond to pr feedback

* tweak test

* update documentation

* destructure port, remove unnecessary imports

* [fix] export correct type

* [feat] expose createNewServer to plugins

* [fix] respond to pr feedback

* todo: add schema validation & integration test

* use reach

* [fix] use validateKey to validate partial

* [fix] change config shadowing

* check kibana port & prevent shadowing

* centralize start/stop for servers, add integration test

* remove unnecessary property

* never forget your await

* remove option to pass config into start

* fix pr feedback

* fix documentation

* fix test failures

renovate[bot] committed:
Configure Renovate (#37385)

* Add renovate.json

* Update and rename renovate.json to renovate.json5

* Update renovate.json5

* Update renovate.json5

* extend the base config

* remove enabled:false

* skip the part limiting managers, we point to specific package.json files instead

* try to disable updating all but eslint packages

* nest npm specific config

* fix syntax error

* bad copy job

* tweak labels

* disable update requests for `@kbn/` packages

* stop overriding rangeStategy

* extend package.json selectors

* add a couple more package.json files

zfy0701 committed:
[Code] change suggested clone url (#37354)

w33ble committed:
Fix: handle null argument values (#37411)

typeof null === 'object', so add an explicit check for null

clintandrewhall committed:
Fix formatdate not allowing strings (#37301)

* Fix formatdate not allowing strings

* Addressing Feedback

zfy0701 committed:
[Code] add util scripts to run tests/checks (#37365)

mw-ding committed:
[Code] Improve git url parsing for github repo url with branches/refs (#37397)

mw-ding committed:
[Code] Make file and repository typeahead query case insensitive (#37399)

FrankHassanabad committed:
[SIEM] Fixed timeline Epoch for zeek to not render in Epoch (#37290)

## Summary
  * Added a few functions to test for Epoch and correct things where they can
  * Added unit tests where there were none
  * Added unit test that exercises the failure (TDD)
  * Fixed the unit test and ensured the other unit tests still run
  * https://github.com/elastic/ingest-dev/issues/280

Before fix:
<img width="1272" alt="broken" src="https://user-images.githubusercontent.com/1151048/58518929-89375000-816e-11e9-858b-21daefc8b19c.png">

After fix:
<img width="1091" alt="Screen Shot 2019-05-28 at 5 29 09 PM" src="https://user-images.githubusercontent.com/1151048/58518937-8d636d80-816e-11e9-9ba8-dfd3fd054416.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~

- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

w33ble committed:
Fix: ensure filter group always shows up in the sidebar (#37419)

* fix: ensure filter group always shows up in the sidebar

fixes #36950

* fix: verify argValue exists before adding to the choices list

Co-Authored-By: Catherine Liu <catherineqliu@outlook.com>

nreese committed:
[Maps] heatmap color ramp select and legend details (#37187)

* [Maps] heatmap color ramp select and legend details

* heatmap style editor

* add tests for color_utils

* add snapshot test for HeatmapStyleEditor

* change label to Color range

* update jest snapshot for label text change

nreese committed:
[Maps] only sync layer list state to mapbox once (#37133)

* [Maps] only sync layer list state to mapbox once

* clean up getDerivedStateFromProps logic

stephmilovic committed:
[SIEM] Bugfix for reading initial URL state of timerange for timeline (#36954)

SIEM URL state bugfix and cleanup

Bargs committed:
fetchIndexPatterns fixes (#37413)

Includes a couple of fixes for the fetchIndexPatterns utility function

Stop quoting search terms
Only call getFromSavedObject once on the default index pattern

Commit:
[Renovate] disable automatic rebase on base branch change (#37443)

XavierM committed:
[SIEM] Escape Query Bug (#37390)

* Fix Escape query bug on timeline query

* just realized with andrewG that updateColumns action is not called as before and also some attribute is not saved anymore in the columns attributes

* cleanup

renovate[bot] committed:
Pin dependencies (#37404)

* Pin dependencies

* [eslint] ignore large files that take a long time to parse

* [eslint-plugin-import] force `querystring` to be considered a core module

* [eslint-plugin-import] auto-fix import order issues

Commit:
Renovate - add package groups for test/task runners (#37455)

Commit:
[renovate] enable rangeStrategy: pin to try avoiding rebases

renovate[bot] committed:
Pin dependencies (#37454)

Commit:
switch to explicitly authorizing renovate issues

https://github.com/renovatebot/config-help/issues/256#issuecomment-497210984

Commit:
[renovate] extend dependency groups

Commit:
fix(code/frontend): repo not exists error page (#36093)

fix(code/frontend): repo not exists error page

renovate[bot] committed:
Pin dependencies (#37406)

* Pin dependencies

* include core-js in babel packages

* specify core-js version when using `useBuiltIns`

* dedupe @babel/types to avoid "range of null" problem

Commit:
[renovate] disable semantic commit style

Commit:
[esArchiver] await esArchiver.load() call

Commit:
[renovate] extend package groups

renovate[bot] committed:
Update dependency mocha to v3.5.3 (#37458)

justinkambic committed:
Add a function to the time picker page object to set a commonly-used timespan option. (#37374)

nreese committed:
[Maps] updated scaled property values when there are no results (#37121)

sorenlouv committed:
[APM] Fix issue with missing agentName on metrics page (#37210)

ppisljar committed:
fixing typo (#37472)

sorenlouv committed:
[APM] Refactor ESClient to allow other operations than `search` (#37334)

joshdover committed:
[new-platform] Update and clarify Migration guide (#36833)

kobelb committed:
Removing the "type" from the deleteByQuery that esArchiver uses (#37507)

* Removing the "type" from the deleteByQuery that esArchiver uses

* [context] remove type from ids filter in context app

monfera committed:
[Canvas] Type strength lock-in for matrix operations (#37516)

Type strength lock-in for matrix operations

jkakavas commented:
Apologies for this mess. My git-fu failed me, I'll close and reopen.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/16215/)
